mend_sca_cleanup_tool/sca_cleanup_tool.py

```diff
@@ -368,22 +416,53 @@ def parse_config_file(filepath):
                     project_parallelism_level=config['DEFAULT'].get('ProjectParallelismLevel', 5),
                     dry_run=config['DEFAULT'].getboolean("DryRun", False),
                     skip_report_generation=config['DEFAULT'].getboolean("SkipReportGeneration", False),
-                    skip_project_deletion=config['DEFAULT'].getboolean("SkipProjectDeletion", False)
+                    skip_project_deletion=config['DEFAULT'].getboolean("SkipProjectDeletion", False),
+                    proxy=config['DEFAULT'].getboolean("proxy", '')
                 )
     else:
         print(f"No configuration file found at: {filepath}")
         exit()
 
-def post_api_request(request):
+
+def post_api_request(request, report = False):
     try:
-        MAIN_API_CONNECTION.request("POST", '/api/v1.4', request, HEADERS)
+        #  Using Http request, this code here just for testing requests API call
+        #r = call_api(header=HEADERS, data=request, report=report)
+        #return r
+        MAIN_API_CONNECTION.request("POST", f'/api/v1.4', request, HEADERS)
         return MAIN_API_CONNECTION.getresponse().read()
-    except:
-        sys.exit(f"There was an issue calling the Mend API with URL: {CONFIG.mend_url}")
+    except Exception as err:
+        print(f"Host: {MAIN_API_CONNECTION.host}")
+        r = call_api(header=HEADERS, data=request, report=report)
+        return r
```

Comment:
Why the try except here? Can't we detect when the proxy is in place and use the correct call instead of waiting for the first call to fail?

The check should happen in a separate method to call either post_api_request or call_api.

Reply:
We can't detect it and I think don't need. The Http.request is good method, just can't be used when you have tunnel or something like that. I wanted to keep the connection mechanism, because don't want to get any another issues or need to remove it from any place and change to requests.request

Reply:
Didn't we detect it at the start with "if MAIN_API_CONNECTION._tunnel_host"? Or "if CONFIG.proxy_host and CONFIG.proxy_port:" or maybe I am not understanding.

I understand you didn't want to touch much. But if calling two different HTTP Library is going to mangle the code, we should do it the right way and move to a single HTTP library.

Reply:
@DimarrWS - Intermingling a method like this is not the correct way to do this, this is a bad design.  We should have a new method for your changes as they have different needs.  Using a new method for your changes provides a clear new path without compromising the existing process.

Reply:
As a side effect of waiting for the first call to fail, we are going to slow the application down. Depending on how the customer configures the script. We could be making thousands of calls that are now doubled waiting for the first one to fail.